upload-api/tables/delegations.js

```diff
@@ -121,36 +121,35 @@ export function useDelegationsTable (dynamoDb, tableName, { bucket, invocationBu
       })
       const response = await dynamoDb.send(cmd)
 
-      const delegations = []
+      const delegationPromises = []
       for (const result of response.Items ?? []) {
         const { cause, link } = unmarshall(result)
         const delegationCid = /** @type {Ucanto.Link} */ (parseLink(link))
         if (FIND_DELEGATIONS_IN_INVOCATIONS && cause) {
           // if this row has a cause, it is the CID of the invocation that contained these delegations
           // and we can pull them from there
           const invocationCid = /** @type {Ucanto.Link} */ (parseLink(cause))
-          const result = await findDelegationInInvocation(
+          delegationPromises.push(findDelegationInInvocation(
             {
               invocationBucket, workflowBucket,
               invocationCid, delegationCid
             }
-          )
-          if (result.ok) {
-            delegations.push(result.ok)
-          } else {
-            return { error: new Failure(`could not find delegation ${delegationCid} from invocation ${invocationCid}`) }
-          }
+          ))
         } else {
           // otherwise, we'll try to find the delegation in the R2 bucket we used to stash them in
-          const result = await cidToDelegation(bucket, delegationCid)
-          if (result.ok) {
-            delegations.push(result.ok)
-          } else {
-            return {
-              error: new Failure(`failed to get delegation ${delegationCid} from legacy delegations bucket`, {
-                cause: result.error
-              })
-            }
+          delegationPromises.push(cidToDelegation(bucket, delegationCid))
```

Comment:
nit: from my experience with R2, and their rate limits, I would suggest wrapping this with a `p-retry` (if error is not 404), or maybe better within the function itself.

Reply:
This might even be the cause for the 500 errors when there are lots to retrieve. We shouldn't block this PR on adding that though but it would be a good idea to have in a follow up ;)